UI/window-basic-main.cpp

```diff
@@ -167,6 +167,117 @@ static int CountVideoSources()
 	return count;
 }
 
+static void MIDICallback(double deltatime, std::vector<uint8_t> *message,
+		void *userData)
+{
+	UNUSED_PARAMETER(userData);
+	QMidiEvent *midiEvent = nullptr;
+	size_t nBytes = message->size();
+
+	if (nBytes > 0) {
+		midiEvent = new QMidiEvent(*message, deltatime);
+		blog(LOG_DEBUG, "MIDI message (%s):",
+				obs_key_to_name(midiEvent->getKey()));
+	} else {
+		return;
+	}
+
+	for (size_t i = 0; i < nBytes; i++) {
+		uint8_t val = message->at(i);
+		blog(LOG_DEBUG, "%i (0x%x)", val, val);
+	}
+
+	blog(LOG_DEBUG, "MIDI timestamp: %f\n", deltatime);
+
+	/* todo: figure out how to post events properly */
```

Comment:
Is this still an active todo?

Reply:
I'm not 100% certain, I think I got this working right*